.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish final artifacts to Maven Central repository
+
+on:
+  push:
+    tags:
+      - armeria-*
+
+jobs:
+  publish:
+    name: Publish final artifacts
+    if: github.repository == 'line/armeria'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - id: setup-jdk-16
+        name: Set up JDK 16
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '16'
+
+      - name: Restore the cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/wrapper
+            ~/.gradle/caches
+          key: build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('gradle.properties', 'gradle/wrapper/gradle-wrapper.properties', '**/build.gradle', 'dependencies.yml', '*/package-lock.json') }}
+          restore-keys: |
+            build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-
+            build-${{ matrix.java }}-${{ runner.os }}-
+
+      - name: Publish and close repository
+        run: |
+          ./gradlew --no-daemon --stacktrace --max-workers=1 publish closeAndReleaseStagingRepository
```

Comment:
Does publish run the tests also? I think normally you would run `build` as a step before publish, unless you're confident in 100% reproducibility of build (guessing not)

Reply:
> Does publish run the tests also? 

`publish` does not depend on `test` task. There is a test job that is triggered when a new commit is pushed to the master branch. 
https://github.com/line/armeria/blob/e294957e2999d361c3a2450ac85a2b54c1d2322b/.github/workflows/actions_build.yml#L3-L5
I thought it was enough for verification. But it makes sense to run `build` for double check.  👍

Reply:
>  But it makes sense to run build for double check. 👍

Then, should we probably publish the artifact when the build succeeds?

Reply:
Yes, it is. It is noted that flaky tests are excluded. I think the build should pass in the self-hosted runners.